lib/dest/index.js

```diff
@@ -12,11 +12,21 @@ function dest(outFolder, opt) {
   }
 
   function saveFile(file, enc, cb) {
+    /* jshint validthis: true */
+    var stream = this;
+
+    function complete(err, file) {
+      cb(err, file);
+      if (!err) {
```

Comment:
Hmm, so http://bit.ly/1kyt00f says that readable will return a boolean which will say if more data should be pushed into the buffer which is controlled by the highWaterMark http://bit.ly/1kytDGX then as it hits false in the case of theres no piping happening we can then `stream.read` to pull chunks again from the buffer http://bit.ly/1kytTpp. When `piping` from a destination `stream.read` never gets called because the `.pipe` handles the draining of the readable part http://bit.ly/1kyvnjr

I've tried my way with a pipe from the `dest` and not piping essentially `finish`ing the write part and it works..